### PR TITLE
Initial implementation of overlay.

### PR DESCRIPTION
**Pull request by ericawright**
note: I started to add a README, but couldn't see any application for it after reading the way the other readmes were used. Let me know if I'm misunderstanding here.

I'm also considering adding a background image, to show that the overlay is **on top** of something.

Should it be interactive, and if so what is the **firefox** way of doing that? I left it since I didn't want to over-complicate it.
ex: click on overlay makes it and the popup close and click on 'x' on popup does the same.

screen shot 2018-11-20 at 5 06 59 pm

### CODE REVIEW

File: components/Overlay/index.css [hunk 1]
```diff
@@ -0,0 +1,17 @@
+.overlay {
+  position: absolute;
+  top: 0;
+  bottom: 0;
+  left: 0;
+  right: 0;
+  background-color: var(--grey-90-a60);
+}
+
+.popup {
```
Comment (lmorchard) [inline components/Overlay/index.css]: I'd move this into the story - could be a separate `stories.css` only imported in `stories.js`. I don't think we want to include this sample content in the distributed build.

File: components/Overlay/index.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import React from "react";
+
+import "./index.css";
+
+export const Overlay = () => (
```
Comment (lmorchard) [inline components/Overlay/index.js]: Probably want to accept a `({ children })` parameter here.

File: components/Overlay/index.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  <div>
+    <div className="overlay">
+      <div className="popup"></div>
```
Comment (lmorchard) [inline components/Overlay/index.js]: Then, use `{children}` here instead of the popup div.

File: components/Overlay/stories.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import React from "react";
+import { storiesOf } from "@storybook/react";
+import { Sample } from "../../lib/stories-common";
+
+import Overlay from "./index";
+
+storiesOf("Overlay", module)
+  .add("Everything", () => (
+    <Sample notes={"This is a simple Overlay, with a popup to illustrate the use of an overlay."}>
+      <Overlay/>
```
Comment (lmorchard) [inline components/Overlay/stories.js]: Finally, after adding support for children in the React component, you can include the popup sample here like so:
```jsx
<Overlay>
  <div className="popup"></div>
</Overlay>
```

File: components/Overlay/stories.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import React from "react";
+import { storiesOf } from "@storybook/react";
+import { Sample } from "../../lib/stories-common";
+
+import Overlay from "./index";
+
```
Comment (lmorchard) [inline components/Overlay/stories.js]: Could add an `import "./stories.css"` here for story-specific CSS (i.e. the example popup)

File: components/Overlay/index.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import React from "react";
+
+import "./index.css";
+
+export const Overlay = () => (
+  <div>
```
Comment (lmorchard) [inline components/Overlay/index.js]: Probably don't need this extra div, here.

**Comment by lmorchard** [comment]
> note: I started to add a README, but couldn't see any application for it after reading the way the other readmes were used. Let me know if I'm misunderstanding here.

I think we're kind of figuring out what the READMEs are for, still. 😞 Could add a stub README with usage doco. But, otherwise, we've got issue #48 to sort out README notions.

> I'm also considering adding a background image, to show that the overlay is on top of something.
> Should it be interactive, and if so what is the firefox way of doing that? I left it since I didn't want to over-complicate it.
> ex: click on overlay makes it and the popup close and click on 'x' on popup does the same.

We're figuring this stuff out still, too. I'm leaning toward not making these components interactive unless absolutely necessary, so we can stay useful with just HTML & CSS. So far, the React components are mostly just fancy markup templates.

For demonstration purposes, you could add this kind of interactivity to the story, though. I do a bit of this in [my PR for Tabs](https://github.com/FirefoxUX/photon-components-web/pull/71/files#diff-154390db19ee92b7f3ad389e707cecb3R16), but I'm still kind of scratching my head whether this is the best way to do.

**Comment by ericawright** [comment]
updated, I decided against making an interactive version for now

**Comment by lmorchard** [comment]
Looks good to me!

**Pull request merged**